instant-xml-macros/src/lib.rs

```diff
@@ -181,22 +211,47 @@ pub fn to_xml(input: TokenStream) -> TokenStream {
 
     serializer.add_footer(&root_name, &mut output);
 
-    TokenStream::from(quote!(
+    let current_prefixes = serializer.keys_set();
+    proc_macro::TokenStream::from(quote!(
         impl ToXml for #ident {
-            fn write_xml<W: ::std::fmt::Write>(&self, write: &mut W) -> Result<(), instant_xml::Error> {
-                write.write_str(&(#output))?;
+            fn serialize(&self, serializer: &mut instant_xml::Serializer, _field_data: Option<&mut instant_xml::FieldData>) -> Result<(), instant_xml::Error> {
+                let mut field_data = instant_xml::FieldData {
+                    field_name: #root_name,
+                    field_attribute: None,
+                };
+
+                // Check if prefix exist
+                #(
+                    if serializer.parent_prefixes.get(#missing_prefixes).is_none() {
+                        return Err(instant_xml::Error::WrongPrefix);
+                    }
+                )*;
+
+                // Adding current prefixes
+                let mut to_remove: Vec<&str> = Vec::new();
```

Comment:
We statically now `current_prefixes` so I don't think we have to create a list with them in order to be able to remove them? The loop below could just replace `to_remove` with an array expansion of the current prefixes.

Reply:
I might not follow, but the `current_prefixe` may hide the `parent_prefix`. This is the reason for remove, you don't want to remove prefix that existed in parent, we only have to remove the added ones. If there is simpler way of doing that then great, but nothing came to my mind.

Reply:
Okay, fair enough. I guess my main concern was allocating a `Vec` for this, I guess we could just create an array here instead (since we know the length of the array at codegen time).

Reply:
We don't, as we don't know which prefixes already exist in parent. We are checking it in the next line.

Reply:
Okay, never mind, we can optimize this later.